File: Olimpiady/OIJ:OIG/||| etap/sejf.cpp
#include <bits/stdc++.h>

using namespace std;
int main()
{
    long long wynik;
    int rozmiar,ilosc_zapytan,a,b;
    char typ;
    cin >> rozmiar >> ilosc_zapytan;
    int kolumny[rozmiar+7];
    int wiersze[rozmiar+7];
    for (int i=0; i<rozmiar+8; i++){
        kolumny[i] = i;
        wiersze[i] = i;
    }
    for (int i=0; i<ilosc_zapytan; i++){
        cin >> typ >> a >> b;
        if (typ == 'W'){
            swap(wiersze[a],wiersze[b]);
        }
        else if (typ == 'K'){
            swap(kolumny[a],kolumny[b]);
        }
        else{
            wynik = ( (long long)rozmiar*(wiersze[a] - 1) + kolumny[b] );
            cout<<wynik<<endl;
        }
    }

    return 0;
}

File: Olimpiady/OIJ:OIG/||| etap/sejf.py
rozmiar,ilosc_zapytan = [int(i) for i in input().split()]
kolumny = [j for j in range(1,rozmiar+1)]
wiersze = [j for j in range(1,rozmiar+1)]
for i in range(ilosc_zapytan):
    typ,a,b = input().split()
    if typ == 'W':
        wiersze[int(a)-1],wiersze[int(b)-1] = wiersze[int(b)-1],wiersze[int(a)-1]
        #print(wiersze)
    elif typ == 'K':
        kolumny[int(a)-1],kolumny[int(b)-1] = kolumny[int(b)-1],kolumny[int(a)-1]
        #print(kolumny)
    else:
        print((wiersze[int(a)-1] - 1) * rozmiar + kolumny[int(b)-1])